parsers/GT.py

```diff
@@ -95,36 +106,51 @@ def fetch_production(
     ]
 
 
-def index_api_data_by_hour(json):
-    """The JSON data returned by the API is a list of objects, each
-    representing one technology type. Collect this information into a list,
-    with the list index representing the hour of day.
-    """
+def _get_api_data(
+    session: Session | None,
+    url: str,
+    date_time: datetime,
+    target: Literal["consumption", "production"],
```

Comment:
I would probably use a string enum instead to reduce the risk of miss spellings in the future. It can also be used as a type and has type hinting so you don't need the literal.